aea/aea_builder.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains utilities for building an AEA."""
+from typing import Dict, Any, Tuple, Set
+
+from aea.aea import AEA
+from aea.configurations.base import PublicId, PackageConfiguration, AgentConfig, ConfigurationType
+from aea.configurations.loader import ConfigLoader
+from aea.crypto.ledger_apis import LedgerApis
+from aea.crypto.wallet import Wallet
+from aea.identity.base import Identity
+from aea.mail.base import Address
+from aea.protocols.base import Protocol
+from aea.registries.base import Resources
+
+PackageId = Tuple[AgentComponentType, PublicId]
+
+
+class _Package:
```

Comment:
This class should live in the base folder imo. Something like `aea/common.py`?

Reply:
don't know, let's see